reminder_batch_scheduler/reminder_batch.py

```diff
@@ -0,0 +1,168 @@
+        print('Print batches:', colored(f'{starting_batch} - {final_batch}', 'red'))
+        print('Current total cases:', colored(total_cases, "red"))
+        if not confirm_insert_rules():
+            print(colored('ABORTING', 'red'))
+            return
+        insert_action_rules(action_rules)
+        print()
+        print(colored("All action rules inserted", 'green'))
+
+
+def count_batch_cases(batch, wave_classifiers):
+    wave_classifiers['print_batch'] = [str(batch)]
+
+    batch_count_query, query_values = build_batch_count_query(wave_classifiers)
+    result = db_helper.execute_parametrized_sql_query(batch_count_query, query_values)
+    return result[0][0]
+
+
+def build_batch_count_query(wave_classifiers):
+    classifier_sql_filters = ['']
+    query_param_values = []
+    for classifier, values in wave_classifiers.items():
+        classifier_sql_filters.append(f" {classifier} IN %s")
+        query_param_values.append(tuple(values))
+    classifiers_query_filters = ' AND'.join(classifier_sql_filters)
+
+    return ("SELECT COUNT(*) FROM actionv2.cases "
+            "WHERE receipt_received = 'f' "
+            "AND address_invalid = 'f' "
+            "AND skeleton = 'f' "
+            "AND refusal_received IS DISTINCT FROM 'EXTRAORDINARY_REFUSAL'"
+            "AND case_type != 'HI'"
+            f"{classifiers_query_filters};"), tuple(query_param_values)
+
+
+def build_action_rule_classifiers(wave, selected_batches):
+    action_rule_classifiers = {}
+    for action_type in constants.ACTION_TYPES_FOR_WAVE[wave]:
+        classifiers = constants.ACTION_TYPE_CLASSIFIERS[action_type]
+        classifiers['print_batch'] = selected_batches
+        action_rule_classifiers[action_type] = json.dumps(classifiers)
+    return action_rule_classifiers
+
+
+def select_batches(starting_batch, wave_classifiers, max_cases):
+    total_cases = 0
+    selected_batches = {}
+
+    for batch in range(starting_batch, 100):
+        batch_case_count = count_batch_cases(batch, wave_classifiers)
+        print(f'Batch: {batch}, Count: {batch_case_count}')
+        if total_cases + batch_case_count > max_cases:
+            break
+        total_cases += batch_case_count
+        selected_batches[str(batch)] = batch_case_count
+
+    return selected_batches
+
+
+def confirm_insert_rules():
+    confirmation = input(
+        colored('WARNING: You are about to write the the action rules to the database, '
+                'resulting in materials being sent for print. \nContinue? [Y/n] ', color='red'))
+    if confirmation != 'Y' and confirmation.lower() != 'yes':
+        return False
+    return True
+
+
+def generate_action_rules(action_rule_classifiers, action_plan_id):
+    action_rules = {}
+    for action_type, classifiers in action_rule_classifiers.items():
+        raw_trigger_date_time = input(f'Input trigger date time for action type {action_type} (RFC3339): ')
+        trigger_date_time = rfc3339.parse_datetime(raw_trigger_date_time)
```

Comment:
having to input the time multiple times would probably be kinda annoying. My personal opinion is that it'll be fine to trigger all the action rules at the same time

Reply:
Fair point but If you want them all at the same time it's easy to paste in the same time a couple of times. I wasn't sure if we've ruled out staggering the rules.  If we're sure they can all go at once I could make it an argument instead.